internal/driver/deviceclient.go

```diff
@@ -177,7 +192,24 @@ func TransformCommandValueToDataBytes(commandInfo *CommandInfo, value *sdkModel.
 	var byteCount = calculateByteCount(commandInfo)
 	var err error
 	var maxSize = uint16(len(value.NumericValue))
-	var dataBytes = value.NumericValue[maxSize-byteCount : maxSize]
+	var dataBytes []byte
+	switch value.Type {
+	case sdkModel.String:
+		oriStr := value.ValueToString()
+		tempBytes := []byte(oriStr)
+		bytesl := len(tempBytes)
+		oribytel := int(commandInfo.Length * 2)
+		if bytesl < oribytel {
+			less := make([]byte, oribytel-bytesl)
+			dataBytes = append(tempBytes, less...)
+		} else if bytesl > oribytel {
+			dataBytes = tempBytes[:oribytel]
+		} else {
+			dataBytes = []byte(oriStr)
+		}
+	default:
+		dataBytes = value.NumericValue[maxSize-byteCount : maxSize]
+	}
```

Comment:
Please refactor this code snippet to the if else block https://github.com/edgexfoundry/device-modbus-go/blob/6f937c290bfa4078b9e6fff95cd67f5638b9ced7/internal/driver/deviceclient.go#L220. 
```
if commandInfo.ValueType == sdkModel.Uint32 || commandInfo.ValueType == sdkModel.Int32 || commandInfo.ValueType == sdkModel.Float32 {
		...
} else if commandInfo.ValueType == sdkModel.string{
   ...
}
```

Reply:
ok, i`ll reformat there , thank you